### DISCUSSION TITLE
Get verified on Mastodon with your Joomla Astroid website?

### DISCUSSION THREAD

**Discussion by paurray** [templaza/astroid-framework#256] [Q&A]
Hello in an age where people are fleeing Twitter and going to among others Mastodon.
I am wondering do I paste the code below into my template on the home page. And if so where?
(tried Google & joomla.org but found nothing.)

http://support.wikilabs.asia/index.php?/News/NewsItem/View/6368/get-verified-on-mastodon-with-your-website

Or am I barking up the wrong tree?

```
>
>
>My website>
>
>
 rel="me" href="https://mastodon.example.com/@tux">Mastodon>
>
>
```

any help greatly appreciated

Paul

**Comment by Chacapamac** [comment]
**I also quit Twitter forever.** _Tired of the bull around that social platform._
I’m interested by that.

I look at your link **and if I understand** this is the way to **insert a direct links** to your Mastodon **and also make sure that your link is Mastodon/Verified.**

I have no idea of the platform, yet...

**On this link they says**

> The verification code you got from your Mastodon profile page is just a hyperlink back to your profile page, around the word "Mastodon." It's arguably the most obvious word to link back to a Mastodon profile, but it's entirely arbitrary. What really matters is that a link to your profile page, with the rel="me" attribute, appears on a page of a website you control. The contents of the link doesn't actually matter.


**They even have Share button on Github**
https://aly-ve.github.io/Mastodon-share-button/

**Mastodon have an API documentation** —> https://docs.joinmastodon.org/api/

**Also _“Embed posts in external websites”_ discussion on Github**
https://github.com/mastodon/mastodon/issues/1065

**Comment by paurray** [comment]
@Chacapamac  I am really enjoying Mastodon experience so far. Very refreshing after Twitter. I know it sounds like a contradiction but it seems more intimate and yet very international at the same time. No BS. Probably has something to do with the Algorithms. And no adverts. Hurah.

Really like this intro here (If Mastodon Existed In The 1980s...): https://www.youtube.com/watch?v=zV8Q0WOYoag

I spend a lot of time in a video post production bubble online. And one of the hashtags there is #postchat. Some one who is very active and well regarded in this online post production bubble set up a server here:

https://postchat.io

There is no doubt something similar for Joomla need to check that out.

By the way if you are still on Twitter there are projects on Github where you can search all your Twitter friends for Mastodon links:
https://twitodon.com/

Any way I signed up and filled out my profile on Mastodon. 

Here is my profile:

<img width="576" alt="image" src="https://user-images.githubusercontent.com/9163424/205714446-50bd773b-430a-435b-88ac-11f8cbeb2adf.png">

01-Mastodon-MyProfile

And this is me here editing my profile:

<img width="1424" alt="image" src="https://user-images.githubusercontent.com/9163424/205714545-cd627393-f78a-416b-8414-2c67ab413aa6.png">

02-Mastodon-Edit-Profile

The object of the exercise is to have the little box on Mastodon where it says my "website" turn green. Please see "MyProfile" above.

The first thing I tired was posting the code in a custom module on my homepage. Probably makes no sense and does not seem to work. Gives me a little link at the bottom of the page which I do not really want.

<img width="960" alt="image" src="https://user-images.githubusercontent.com/9163424/205714665-3804d8a4-74f8-4682-a9ad-f763f6756722.png">

03-Mastodon-Edit-Home-Page

I also went into the Astroid Template options into the Custom code section and tried adding the code here. Also probably does not make sense I know. And does not seem to work any way. Apparently these things can also take some time to ripple through. Have been trying off and on for a couple of weeks now.

<img width="960" alt="image" src="https://user-images.githubusercontent.com/9163424/205714792-ae6838c1-05d9-4a55-bb9f-c41cdd31e18d.png">

04ba-Mastodon-Custom-Code-FrontEnd

<img width="960" alt="image" src="https://user-images.githubusercontent.com/9163424/205714866-9c647ee6-8ff9-4ac3-b533-9fe497e3e71a.png">

04b-Mastodon-Custom-Code-BackEnd

I guess at the end of the day if you can post code from Mastodon into a site the chances are that you own it and this is the way your ownership is verified. Hope this makes sense? Kind of like a DIY check mark that does not cost 8$ a pop and works/mostly ;-)

**Comment by Chacapamac** [comment]
**Thank you for the info Paul.**
One thing, I don’t think you should use the **Custom Code > Tracking Code** as it is specially made for Google Analytics code insertion.

I think, I understand what you want to do. 
You want to be able to show a link for your Mastodon profile in a way that its easy to use around your website.

• The first thing I will try is to add the link directly to the **Social Icons** in **Astroid Layout Manager** to see if it can work there
— You can probably CSS this how you want
— You also ca use the Awesome Icon for Mastodon —> <i class="fab fa-mastodon"></i>

![Screen Shot 2022-12-06 at 11 24 34 AM](https://user-images.githubusercontent.com/1907276/205970386-3681f3bb-13fa-4099-8314-53f23115cec3.jpg)

• **After I probably like to have a way to insert it anywhere in the website easily.**
— I will try to make a module with only the Mastodon link, with or without surrounding HTML to CSS it the way you want, or with the Awesome icon to display some kind of button in articles or in a module positions.

— For me, my favorite way will probably to use my trusty “Snippets” _(the free one is good)_ from the trusty **RegularLab Dev** —> https://regularlabs.com/snippets
— Here you can fabricate you Mastodon “Snippet” code and just use the insert snippet code to put your button anywhere you want in Joomla

I Forget that If you have a good editor (I use JCE PRO) you can create “Code Snippets” and use them in your edition. Thats maybe better than add another components/plugins

**Comment by paurray** [comment]
Hello @Chacapamac 

thanks for your great suggestions. Gratefully appreciated.
Yes I have JCE Pro & just installed Snippets from the most wonderful Regular Labs site. Thanks for the link.

I will do digging & some testing & get back to you.

Here is I think what is supposed to happen as in the Wordpress Universe (Could not find anything for Joomla) :

https://opensource.com/article/22/11/verified-mastodon-wordpress

And what you are suggesting

> “You want to be able to show a link for your Mastodon profile in a way that its easy to use around your website.”

Is also very cool.

More soooooonish. End of year madness kicking in here already. 

thanks again

Paul

**Comment by paurray** [comment]
Hi @Chacapamac 

On a side note I found this whilst doing some research. Might be interesting to some:

https://github.com/nikosdion/fediverse/releases/tag/1.0.0

(Joomla! extensions to integrate Mastodon with your site) from Nicholas K. Dionysopoulos released.

I did as suggested. Am posting details here in case it helps anyone else. Not an expert so forgive me if I get some of this wrong.

A) Setting up Social Icons

- As suggested by Chacapamac I went to the Social Tab in the Astroid Template back end.

<img width="540" alt="01-Set-Up-Mastodon-Profile" src="https://user-images.githubusercontent.com/9163424/207296080-eeb968f5-973e-4c40-8f78-bf2be5ffee4c.png">

- If it is not yet active make sure that the Social Features are switched on by sliding the tiny Social activation slider!

- In the “Link” section I entered https://THE NAME OF MY MASTODON SERVER/@MY USER NAME

- I found the following code on the Font Awesome site and entered it in the icon class section.

fa-brands fa-mastodon

https://fontawesome.com/icons/mastodon?s=solid&f=brands

This gives me the Mastodon symbol. 

- Next I saved my changes.

<img width="540" alt="02-Set-Module-Position" src="https://user-images.githubusercontent.com/9163424/207296256-ac22e967-07c8-4a01-adc9-c68dfe2c3dbd.png">

- Set in my case Module Position and published to “Astroid Top Social”

B) Getting Icons to show and format

- Used this to help me get icons to show and understand formatting.

https://docs.joomla.org/Module_Position

Setting the Position of the Social Icons

- I went to the Layout Section of the Astroid Template and created a new Content Section with a single row/colum. And Positioned it in my case between the Header Section and the Content Section. Called it Social Icon Container.

- In the newly created Social Icon Container I added an element by pressing the tiny little plus sign.

- I selected the Module option.

<img width="540" alt="03 Social Icons HERE" src="https://user-images.githubusercontent.com/9163424/207296353-5e6fd544-04e2-4834-a129-ce51510ffb35.png">

- Searching for and added the “Social Icons Module and added the Astroid Social Icons Module position to it. (Some times clearing of cash helps from here on in…) Saved.

<img width="540" alt="04-Social-Icons" src="https://user-images.githubusercontent.com/9163424/207296469-44920fc6-ca16-4578-bbcf-bc48d8cd5922.png">

Astroid Social Icons Module inside a section inside an Element in a column. With some formatting added to the Column settings of the section. I added the formatting to get some spacing and move the icons over to the right. (Not surre if this is the most eloquent way to format but it works for now)

<img width="540" alt="05-Social-Media-Icons" src="https://user-images.githubusercontent.com/9163424/207296585-70994ea8-f740-4264-83d4-364fc2df6465.png">

C) Final Thoughts...

I still have not got the Mastodon verification working as such at the Mastodon end.

`
<a rel="me" href="https://MY SERVER/@MY “USER NAME>Mastodon</a>`

I have added this code multiple times to multiple published modules. As far as I can tell I do not even need this part

`>Mastodon<`

I believe

`>.< `

would do the trick as well and would be 99.9% invisible.


Maybe I am missing some thing but I would have thought that inserting this or the icons as outline above would have resulted in the red box around my site as in this random eg here from the same Mastodon server where I am hosted… If I find a solution I will post back.

<img width="168" alt="06-Random EG" src="https://user-images.githubusercontent.com/9163424/207296674-acd57678-09fb-49a4-8bc9-672272993c65.png">

**Comment by Chacapamac** [comment]
Thank @paurray for the infos.

**I look at Nicholas K. Dionysopoulos Giithub, this is really great**
**_From Nicholas:_**
The extensions in this package help you integrate Mastodon with your site. Currently, the following extensions are included:

**[Fediverse - Mastodon Feed (Module)**]
https://github.com/nikosdion/fediverse/blob/1.0.0/docs/mod_fediversefeed.md. 
**Displays the public Mastodon feed of a user**. 
Supports media (images and videos), as well as content warnings on the entire toot or just each media item.

and
**Content - Embed Toot (Plugin)**
https://github.com/nikosdion/fediverse/blob/1.0.0/docs/plg_content_fediverse.md). 
A plugin to embed toots in articles, Custom HTML modules, and third party extensions which support Joomla's standard content events.

**My Input on Nicholas** 
He is the developer of the fantastic Joomla software **Akeeba Backup** and **Akeeba Admin** that are necessary in any serious Joomla website. Nicholas is also a **BIG reason for the greatness of Joomla 4**. Anything from Nicholas, you can be sure it is top notch to use...
—————————————————————————————————

@paurray , I read your last post and I have some questions

**• You said** 
> I still have not got the Mastodon verification working as such at the Mastodon end.

**Q1.** 
**Are the link working?** 

About the verification, I find this —> https://mashable.com/article/how-to-get-verified-mastodon
(I should open my account on Mastodon soon....)

**Interesting infos**
So, if you put a link in your Mastodon profile, the company "checks if the linked page links back to your Mastodon profile. If so, you get a verification checkmark next to that link, since you are confirmed as the owner." The company does this by checking [the rel="me" attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel) — a standard way to check whether or not linked website belongs to a user on a third party site.

Some tips Mastodon offers to pass verification are making sure the link you're adding is served securely (over HTTPS) and to host Mastodon on your own domain.


**Q2.** 
You said:

> Maybe I am missing some thing but I would have thought that inserting this or the icons as outline above would have resulted in the red box around my site as in this random eg here from the same Mastodon server where I am hosted…

**I’m not sure what you expected to see on your website and what we supposed to see in your example?**

**Comment by paurray** [comment]
Hello @Chacapamac, thank you for your fast response.

Nicholas is obviously a legend. I have had a couple of brief online encounter with him and found him to be no nonsense, honest and incredibly generous. If I am not mistaken he is also behind many of the new multi factor authentication features in Joomla 4.2 + which really speaks volumes for him.

As a matter of interest in case you missed it here is the instance where Nicolas is:

https://fosstodon.org/explore
https://fosstodon.org/@nikosdion

So if you were for eg to sign up there you would be in excellent company. I believe that you can switch instance pretty much seamlessly later if you want taking followers but not posts with you.

How To Mastodon - Pt 3 - Get Verified, Move Instances
https://www.youtube.com/watch?v=aiXYu-Zz38c

With respect to Q1

The link is not working for me as in I do not get the green box on my Mastodon profile.

YOU SAID:

> Some tips Mastodon offers to pass verification are making sure the link you're adding is served securely (over HTTPS) and to host Mastodon on your own domain.

- I have https
- Am not self hosting
- Having done more research I see that people with the Green box on my server are hosted on other larger instances.
- Need to double check

eg:

When logged in to the server I am on as @paurray I see this guy:

https://postchat.io/@stooovie@mas.to

But going to a different browser (not logged in) the above link takes me here:

https://mas.to/@stooovie

<img width="960" alt="Eg-Profile-Server-Other-Server" src="https://user-images.githubusercontent.com/9163424/207559993-b51b82ae-2590-4331-bbb9-88dc76907ed6.png">

- I have not found any one with a green box with the same host as me.
- Perhaps nobody really cares as the group is small and most people know each other or some one who knows someone.
- Maybe it is a limitation of the group where I am being hosted??? Will try to find out more.

With respect to Q2

First my apologies I talked about a red box I meant to say green box. It was a brain fart. Sorry.

I have done some more digging and found an example of a random person where this works as expected on a Joomla site.

https://fimidi.com/@viviana

here you will see the link to her site with the green box.

Viewing the source code I find the following line of code related to the Mastodon link back to Mastodon.

`<p><a href="https://fimidi.com/@viviana" target="_blank" rel="me nofollow noopener" title="Link zu Mastodon">Mastodon</a></p>`

*Where “Link zu” sounds to me like german meaning “link to”

- I suspect that she is self hosting as you suggested this also helps.

- Algain maybe my server just ?does not support? this link confirmation thing at the moment.

I will try to find out more.

**Comment by Chacapamac** [comment]
Well, me to, I want to open a Mastodon account. I’m trying to understand the principle of the multiple server as they seem to have different interests, looking at it....

**Comment by paurray** [comment]
For me at least it was relatively easy. As some one in the online community I frequent actively set up and promoted a #postchat server. As in he was able to use the hashtag #postchat to funnel people onto Mastodon https://postchat.io/ he set up. Maybe search for hashtags that interest you or people who you know who have interesting things to say. For me so far the Mastodon experience has been priceless. I am bookmarking all kinds of stuff and in a sense it is becoming somewhat of a mind map for me.

Also remember you can change later from one server to the other. This is described in a video I linked above.

As a matter of interest I know now that the Mastodon “website verification” is working for other people on the server that I am on. So I am missing something. Will find some time to do more digging from next week onwards…

**Comment by Chacapamac** [comment]
@paurray 
Just to inform you that **Nicholas K. Dionysopoulos** release a new version of it’s Mastodon integration Plugin. Didn’t test is myself??

https://github.com/nikosdion/fediverse/releases/tag/2.0.0
